### PR TITLE
Fix MCP pause case and re-organize drain controller package

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

File: controllers/helper_test.go [hunk 1]
```diff
@@ -1,330 +0,0 @@
-/*
```
Comment (zeeke) [inline controllers/helper_test.go:1]: Why remove this file?

File: controllers/drain_controller.go [hunk 2]
```diff
@@ -250,93 +169,17 @@ func (dr *DrainReconcile) getObject(ctx context.Context, req ctrl.Request, objec
 	return true, nil
 }
 
-func (dr *DrainReconcile) ensureAnnotationExists(ctx context.Context, object client.Object, key string) (string, error) {
+func (dr *DrainReconcile) ensureAnnotationExists(ctx context.Context, object client.Object, key string) (string, bool, error) {
 	value, exist := object.GetAnnotations()[key]
 	if !exist {
 		err := utils.AnnotateObject(ctx, object, constants.NodeStateDrainAnnotationCurrent, constants.DrainIdle, dr.Client)
```
Comment (zeeke) [inline controllers/drain_controller.go:175]: this should be 
```
err := utils.AnnotateObject(ctx, object, key, constants.DrainIdle, dr.Client)
```
right?

**Review by zeeke — COMMENTED** [review]
Just a few minor comments, otherwise LGTM

**Review by ykulazhenkov — COMMENTED** [review]
I checked the PR and I don't have meaningful comments. I think we are good to merge it when comments from Andrea are addressed.

File: controllers/helper_test.go (hunk 1 above)
Comment (SchSeba) [reply controllers/helper_test.go]: this was related to the device plugin one I rebase this PR

File: controllers/drain_controller.go (hunk 2 above)
Comment (SchSeba) [reply controllers/drain_controller.go]: right nice catch!

**Review by zeeke — APPROVED** [review]
LGTM!

**Pull request merged**